ScoutSuite/providers/aws/services/ec2/service.py

```diff
@@ -0,0 +1,35 @@
+from ScoutSuite.providers.base.configs.resources import Resources
+from ScoutSuite.providers.aws.configs.regions_config import Regions, ScopedResources
+from ScoutSuite.providers.aws.services.ec2.ami import AmazonMachineImages
+from ScoutSuite.providers.aws.services.ec2.vpcs import Vpcs
+from ScoutSuite.providers.aws.services.ec2.snapshots import Snapshots
+from ScoutSuite.providers.aws.services.ec2.volumes import Volumes
+
+from opinel.utils.aws import get_aws_account_id
+from ScoutSuite.utils import get_keys, ec2_classic
+
+
+# TODO: Add docstrings
+
+class EC2(Regions):
+    def __init__(self):
+        super(EC2, self).__init__('ec2')
+
+    async def fetch_all(self, credentials=None, regions=None, partition_name='aws'):
+        await super(EC2, self).fetch_all(chosen_regions=regions, partition_name=partition_name)
+
+        # TODO: Is there a way to generalize this? 
```

Comment:
From what I'm seeing here in your PR, I have the feeling that we share a lot of things in the structure of our providers (aws/azure). So I tried to generalize your `ScopedResources` class in my PR (I renamed your class `SimpleResources`) to extend our composite pattern and use a parent/children structure (see https://github.com/nccgroup/ScoutSuite/pull/197/files).

So if you define a `AwsSimpleResources` and a `AwsCompositeResources` like I did for Azure (https://github.com/nccgroup/ScoutSuite/blob/dcf6e3a0196de64c5aaf4ee9864cb5189c47b81c/ScoutSuite/providers/azure/resources/resources.py), I think you could rewrite this `EC2` class to something like:

```python
class EC2(AwsCompositeResources):

    children = [
        Vpcs,
        Snapshots,
        Volumes,
        AmazonMachineImages
    ]
    def __init__(self):
        super(EC2, self).__init__('ec2')

    async def fetch_all(self, credentials=None, regions=None, partition_name='aws'):
        await super(EC2, self).fetch_all(chosen_regions=regions, partition_name=partition_name)
        for region in self['regions']:
            await self.fetch_children(
                parent=self['regions'][region],
                region=region
            )
             # There could be a way to 'generalize' the following lines too, through a callback system but
            # let's keep it simple for now (and see if such system could be useful, after migrating other AWS services):
            self['regions'][region]['instances_count'] = sum([vpc['instances'].count for vpc in self['regions'][region]['vpcs'].values()])
            self['regions'][region]['security_groups_count'] = sum([vpc['security_groups'].count for vpc in self['regions'][region]['vpcs'].values()])
            self['regions'][region]['network_interfaces_count'] = sum([vpc['network_interfaces'].count for vpc in self['regions'][region]['vpcs'].values()])
```